### DISCUSSION TITLE
Not able to connect to ADB wirelessly

### DISCUSSION THREAD

**Discussion by wbwxshao** [yume-chan/ya-webadb#622] [General]
I followed the steps mentioned here https://github.com/yume-chan/ya-webadb/discussions/245#discussioncomment-384030 to try to use WebADB over WiFi. I have the WebSocket.js installed and used the command 
`node .\websockify.js 15555 localhost:38757`
to start. And in the WebADB end (https://tango-adb.github.io/old-demo/) as stated I waited for about 2 mins but the web address didn't show up. So I manually clicked the add button and selected "WebSocket" and then give the address as ws://localhost:15555. But then almost immediately I got the prompt from the brower with the error: 
```
Error: Connection closed unexpectedly
    at https://tango-adb.github.io/old-demo/_next/static/chunks/pages/_app-b8a1688d9ffda068.js:10:9808
```

And in Terminal I got the error:
```
PS C:\Users\wbwxs\Documents\webadb\websockify-js\websockify> node .\websockify.js 15555 localhost:38757
WebSocket settings:
    - proxying from :15555 to localhost:38757
    - Running in unencrypted HTTP (ws://) mode
/
 ::1: WebSocket connection
 ::1: Version undefined, subprotocol:
 ::1: target connection error
 ::1: WebSocket client disconnected: 1005 []
/
```

So any idea why this will happen?

Thanks in advance!!

**Comment by yume-chan** [comment]
> ````
>  ::1: target connection error
> ````

Your WebSockify program can't connect to `localhost:38757`. Make sure the address is correct (usually it's your phone's IP and wireless debugging port).

**Reply by wbwxshao** [reply]
Yeah I checked that, it's matching the address and port listed in the "Wireless Debugging" section on my phone. Is there anything else that could cause this error?

**Reply by yume-chan** [reply]
Your device shows `localhost` in wireless debugging page? Can you show me a screenshot?

The issue exists between WebSockify and your device. Can your PC connect to your phone, for example using `ping`?

**Reply by wbwxshao** [reply]
Not localhost, but it will show the ip address and port to use the wireless connection.
<img width="288" alt="image" src="https://github.com/yume-chan/ya-webadb/assets/44184324/a86896a6-7cb6-4bee-b088-51ad08f1392f">
Yes I can use adb connect to connect to the phone but somehow WebSockify is not working

**Reply by wbwxshao** [reply]
Update:
I made a mistake when using WebSocket, I should use something like: node .\websockify.js 15555 192.168.2.240:38995 instead of using localhost:port number. Now there is something changed with websockify end saying "connected to target", but didn't see any authorizaiton prompt on the device
<img width="980" alt="image" src="https://github.com/yume-chan/ya-webadb/assets/44184324/2c4df8e7-20da-414d-8843-f349027df7aa">
<img width="772" alt="image" src="https://github.com/yume-chan/ya-webadb/assets/44184324/6b04026b-f8cb-4042-aa0f-09f0ab23ec2e">

@yume-chan Could you please check and advise here? Thanks!

**Reply by yume-chan** [reply]
Android 11 wireless debugging doesn't use the old authorization prompt. The client must be already authenticated on USB connection, or using `adb pair` to get authorized (Tango doesn't support the protocol of `adb pair`). So if you have connected to that device in Tango over USB, it should also work when connected wirelessly (I haven't tested this).
